shared/actions/chat/index.js

```diff
@@ -376,58 +376,59 @@ const getThreadNonblockSagaMap = (yourName, yourDeviceName, conversationIDKey, a
+    return
+  }
 
-    const untrustedState = yield select(state => state.entities.inboxUntrustedState.get(conversationIDKey))
+  const untrustedState = yield select(state => state.entities.inboxUntrustedState.get(conversationIDKey))
+
+  // only load unboxed things
+  if (!['unboxed', 'reUnboxing'].includes(untrustedState)) {
+    console.log('Bailing on not yet unboxed conversation', untrustedState)
+    return
+  }
+
+  const rekeyInfoSelector = (state: TypedState, conversationIDKey: Constants.ConversationIDKey) => {
+    return state.chat.get('rekeyInfos').get(conversationIDKey)
+  }
+  const rekeyInfo = yield select(rekeyInfoSelector, conversationIDKey)
+
+  if (rekeyInfo) {
+    console.log('Bailing on chat due to rekey info')
+    return
+  }
 
-    // only load unboxed things
-    if (!['unboxed', 'reUnboxing'].includes(untrustedState)) {
-      console.log('Bailing on not yet unboxed conversation', untrustedState)
+  const oldConversationState = yield select(Shared.conversationStateSelector, conversationIDKey)
+
+  let next = null
+  if (oldConversationState && !action.payload.onlyNewerThan) {
+    if (action.payload.onlyIfUnloaded && oldConversationState.get('isLoaded')) {
+      console.log('Bailing on chat load more due to already has initial load')
       return
     }
 
-    const rekeyInfoSelector = (state: TypedState, conversationIDKey: Constants.ConversationIDKey) => {
-      return state.chat.get('rekeyInfos').get(conversationIDKey)
+    if (oldConversationState.get('isRequesting')) {
+      console.log('Bailing on chat load more due to isRequesting already')
+      return
     }
-    const rekeyInfo = yield select(rekeyInfoSelector, conversationIDKey)
 
-    if (rekeyInfo) {
-      console.log('Bailing on chat due to rekey info')
+    if (oldConversationState.get('moreToLoad') === false) {
+      console.log('Bailing on chat load more due to no more to load')
       return
     }
 
-    const oldConversationState = yield select(Shared.conversationStateSelector, conversationIDKey)
-
-    let next = null
-    if (oldConversationState && !action.payload.onlyNewerThan) {
-      if (action.payload.onlyIfUnloaded && oldConversationState.get('isLoaded')) {
-        console.log('Bailing on chat load more due to already has initial load')
-        return
-      }
-
-      if (oldConversationState.get('isRequesting')) {
-        console.log('Bailing on chat load more due to isRequesting already')
-        return
-      }
-
-      if (oldConversationState.get('moreToLoad') === false) {
-        console.log('Bailing on chat load more due to no more to load')
-        return
-      }
-
-      next = oldConversationState.get('paginationNext', undefined)
-    }
+    next = oldConversationState.get('paginationNext', undefined)
+  }
 
-    yield put(Creators.loadingMessages(conversationIDKey, true))
+  yield put(Creators.loadingMessages(conversationIDKey, true))
 
+  try {
```

Comment:
Move this hear to avoid a spurious `Creators.loadingMessages(conversationIDKey, false)` if we bail out early